Feature/614 preset management

jofemodo opened the pull request.

riban-bw committed:
Implement preset bank management - wip

riban-bw committed:
Fix rename bank

riban-bw committed:
Fix Favourites bank not showing after adding first preset to favourites.
Added whitespace around some operators.

riban-bw committed:
Merge branch 'testing' into feature/614_preset_management

riban-bw committed:
Fixes some issues with bank management

riban-bw committed:
Merge branch 'testing' into feature/614_preset_management

jofemodo committed:
Merge branch 'masterfx' into feature/614_preset_management

jofemodo commented on zyngui/zynthian_gui_bank.py:
```diff
@@ -71,7 +71,8 @@ def select_action(self, i, t='S'):
 
 		# If there is only one preset, jump to instrument control
-		if len(self.zyngui.curlayer.preset_list)<=1 and self.zyngui.curlayer.preset_list[0][0]=="":
+		#TODO: This is inconsistent behaviour which may confuse users
```
It's not inconsistent, it was wrongly commented ;-D
It should be "If the only preset is the empty preset (aka, there is no presets), jump to instrument control"

riban-bw replied:
I have tweaked the comment in my next commit. Also the code is prone to exception. If len == 0 then accessing [0] will error. I have changed code to be safe in next commit.

riban-bw committed:
Fix wrong bank highlighted when saving preset.
Removed "Remove Bank" option. Only "Delete Bank" used.

riban-bw committed:
Merge branch 'feature/614_preset_management' of github.com:zynthian/zynthian-ui into feature/614_preset_management

riban-bw committed:
Fixed deleting preset.
Deleting last preset in bank now closes preset list.
Fixed error if select bank with single empty preset.

riban-bw committed:
Fix empty preset list (first time) after deleting bank.
Don't create duplicate presets.
Remove "Remove Bank" code.
Confirm overwrite preset.
Don't skip preset or bank lists if there is at least one entry. (Previously skipped if only one entry.)
